### DISCUSSION TITLE
Tool panel layout proposal --> new add-on

### DISCUSSION THREAD

**Discussion by EdoFro** [freeplane/freeplane#1304] [Ideas]
Hi,
I wanted to ask for a change in the ToolPanel layout and how it works, so I was about to create a new Feature Request for it.

My idea was to change the ToolPanel to have the Tabs on the right, not on top, and that if you clicked on a tab that is already selected it should minimize the panel.

There are other software that work this way and I always have found this feature very practical: the labels are visible, but the content is only visible when the label is selected.

The thing is that I was not clear myself about the features of this request so I started playing with a script and I got some results and I ended creating an add-on that does what I wanted.

The add-on acomplishes my requierements, so I don't need to post a Feature Request anymore, but maybe it's interesting to others to use, so here it is the link to its repository:

https://github.com/EdoFro/Freeplane_TabbedPanelMod

> This add-on changes the tabbed panel (Tool panel) layout.
> 
> It places its tabs to the right and changes the labels to icons.
> 
> ![Tabs to the right](https://github.com/EdoFro/Freeplane_TabbedPanelMod/raw/main/resources/TabsToTheRight.png)

> After clicking again on the selected tab.
> Content is hidden
> Tabs are still visible
> ![minimizedClickedAgain.png](https://github.com/EdoFro/Freeplane_TabbedPanelMod/raw/main/resources/minimizedClickedAgain.png)

Addon can be downloaded from its [releases page](https://github.com/EdoFro/Freeplane_TabbedPanelMod/releases/)

Best regards,

edo

**Comment by macmarrum** [comment]
This is so cool :sunglasses:

And it also will make using the panel much easier for beginners :rocket: 

It would be great to also disable the edge/triangle that normally opens and closes the panel, to avoid accidental hiding of the icon-buttons.

**Reply by EdoFro** [reply]
> It would be great to also disable the edge/triangle that normally opens and closes the panel, to avoid accidental hiding of the icon-buttons.

I don't know how to do that, but also, I like to have the possibility to hide the panel completely. I don't think it would be such a problem. Let's wait and see.

BR. edo

**Comment by EdoFro** [comment]
I have to add that when using the add-on, the user can define the panels width for each tab independently.

If the user drags the border of the panel, the new width is stored for the selected tab only.

**Reply by macmarrum** [reply]
This is interesting. I didn't expect it, but I can see how it can be useful :+1:

I notices that it's somewhat unreliable for me, i.e. the width sometimes stays unchanged when switching to a different tab or closing/reopening the panel.

**Reply by EdoFro** [reply]
the logic is:
- when starting, all tabs has no width defined
- if you change the width of the panel, it gets stored as the width of the selected tab
- if you change to a tab that has no width stored yet, it continues with same width, but doesn't store this width to that tab.

For example, if you change the width from tab 1 and then go to tab 3 and change the width, both of them will have their own width value.
now if you select tab 2 after tab 1 or tab 2 after tab 3, tab 2 will show different widths each time, because it has no own width defined and it stays with the width from the tab that was previously selected.

I think I'm going to change this logic, because I feel it doesn't feel very natural.

I think it will be:
- when starting, all tabs has no width defined, but the selected one that gets the current width as its own.
- if you change the width of the panel, it gets stored as the width of the selected tab
- if you change to a tab that has no width stored yet, it will get the current width as its initial width.

**Reply by EdoFro** [reply]
logic changed in new version

**Comment by Illegit** [comment]
As a user: Nice! Thanks.

**Comment by Illegit** [comment]
Hi EdoFro, is ti possible to add a functionality that let start FP with the addon already set on?
And not to have to run the script to get it every time you run FP?

**Reply by EdoFro** [reply]
yes, it is.

and I did it 😉 

[Released v0.3](https://github.com/EdoFro/Freeplane_TabbedPanelMod/releases/tag/v0.3) 

BR

edo

**Reply by EdoFro** [reply]
I also created a [topic about the init script](https://github.com/EdoFro/Freeplane_TabbedPanelMod/discussions/3), because I added an option in the preference's dialog for the user to decide if it has to run automatically or not, and it also looks if the add-on is active (the user can inactivate any addon in the tools menu)

I think it may be interesting to others.

**Comment by EdoFro** [comment]
Here is also a little [video](https://github.com/EdoFro/Freeplane_TabbedPanelMod/discussions/4#discussion-5461512) of the addon in action

**Comment by Illegit** [comment]
Working nicely. Thanks.
